query-engine/connectors/sql-query-connector/src/filter_conversion.rs

```diff
@@ -57,13 +57,44 @@ impl Alias {
     }
 }
 
+#[derive(Clone)]
+pub struct ConditionState {
+    reverse: bool,
+    alias: Option<Alias>,
+}
+
+impl ConditionState {
+    fn new(alias: Option<Alias>, reverse: bool) -> Self {
+        Self { reverse, alias }
+    }
+
+    fn invert_reverse(self) -> Self {
+        Self::new(self.alias, !self.reverse)
+    }
+
+    fn alias(&self) -> Option<Alias> {
+        self.alias
+    }
+
+    fn reverse(&self) -> bool {
+        self.reverse
+    }
+}
+
 pub trait AliasedCondition {
     /// Conversion to a query condition tree. Columns will point to the given
     /// alias if provided, otherwise using the fully qualified path.
     ///
     /// Alias should be used only when nesting, making the top level queries
     /// more explicit.
-    fn aliased_cond(self, alias: Option<Alias>, reverse: bool) -> ConditionTree<'static>;
+    fn aliased_cond(self, state: ConditionState) -> ConditionTree<'static>;
+
+    fn aliased_condition_from(&self, alias: Option<Alias>, reverse: bool) -> ConditionTree<'static>
```

Comment:
Do we still need this new interface after the changes in Quaint?

Reply:
I'm always against for adding code that we do not need right now. It adds complexity and complexity means confusion and bugs.